test/e2e/tests/single_zone_e2e_test.go

```diff
@@ -294,6 +302,165 @@ var _ = Describe("GCE PD CSI Driver", func() {
 		}()
 	})
 
+	It("Should create CMEK key, go through volume lifecycle, validate behavior on key revoke and restore", func() {
+		ctx := context.Background()
+		Expect(testContexts).ToNot(BeEmpty())
+		testContext := getRandomTestContext()
+
+		controllerInstance := testContext.Instance
+		controllerClient := testContext.Client
+
+		p, z, _ := controllerInstance.GetIdentity()
+		locationID := "global"
+
+		// The resource name of the key rings.
+		parentName := fmt.Sprintf("projects/%s/locations/%s", p, locationID)
+		keyRingId := "gce-pd-csi-test-ring"
+
+		// Create KeyRing
+		ringReq := &kmspb.CreateKeyRingRequest{
+			Parent:    parentName,
+			KeyRingId: keyRingId,
+		}
+		keyRing, err := kmsClient.CreateKeyRing(ctx, ringReq)
+		if !gce.IsGCEError(err, "alreadyExists") {
+			getKeyRingReq := &kmspb.GetKeyRingRequest{
+				Name: fmt.Sprintf("%s/keyRings/%s", parentName, keyRingId),
+			}
+			keyRing, err = kmsClient.GetKeyRing(ctx, getKeyRingReq)
+
+		}
+		Expect(err).To(BeNil(), "Failed to create or get key ring %v", keyRingId)
+
+		// Create CryptoKey in KeyRing
+		keyId := "test-key-" + string(uuid.NewUUID())
+		keyReq := &kmspb.CreateCryptoKeyRequest{
+			Parent:      keyRing.Name,
+			CryptoKeyId: keyId,
+			CryptoKey: &kmspb.CryptoKey{
+				Purpose: kmspb.CryptoKey_ENCRYPT_DECRYPT,
+				VersionTemplate: &kmspb.CryptoKeyVersionTemplate{
+					Algorithm: kmspb.CryptoKeyVersion_GOOGLE_SYMMETRIC_ENCRYPTION,
+				},
+			},
+		}
+		key, err := kmsClient.CreateCryptoKey(ctx, keyReq)
+		if !gce.IsGCEError(err, "alreadyExists") {
```

Comment:
should it already exist if we're generating uuid every time?

Reply:
no. I guess it should be a hard failure. this code exists from before I was creating UUID's or cleaning up crypto keys